src/main/java/com/google/devtools/build/lib/remote/http/HttpUploadHandler.java

```diff
@@ -138,8 +143,41 @@ private HttpRequest buildRequest(String path, String host, long contentLength) {
     return request;
   }
 
-  private HttpChunkedInput buildBody(UploadCommand msg) {
-    return new HttpChunkedInput(new ChunkedStream(msg.data()));
+  private HttpChunkedInput buildBody(UploadCommand msg) throws IOException {
+    if (msg.casUpload()) {
```

Comment:
How long does it take to do this? Might this trigger the idle timeout?

Reply:
Good question, will investigate. Has the timer started already at this point?

Reply:
It does, yes, although after a much longer time than expected. I added a 10s sleep and set the remote timeout to 1s, this fired one timeout for a static archive after compression but before upload. Setting this to a 2s sleep did not trigger. Need to investigate whats going on here.